store/postgres/src/chain_store.rs

```diff
@@ -2094,12 +2250,15 @@ impl ChainStoreTrait for ChainStore {
                 let this = self.clone();
                 let lookup_fut = async move {
                     let res = this.blocks_from_store(hashes).await;
-                    BlocksLookupResult(Arc::new(res))
+                    BlocksLookupResult::ByHash(Arc::new(res))
                 };
                 let lookup_herd = self.lookup_herd.cheap_clone();
                 let logger = self.logger.cheap_clone();
-                let (BlocksLookupResult(res), _) =
-                    lookup_herd.cached_query(hash, lookup_fut, &logger).await;
+                //TODO(krishna): Add comments explaining the return value of cached_query
+                let res = match lookup_herd.cached_query(hash, lookup_fut, &logger).await {
+                    (BlocksLookupResult::ByHash(res), _) => res,
+                    _ => unreachable!(),
```

Comment:
is this actually unreachable? Probably an error would be better just so we don't panic in case of a logic error

Reply:
Its actually unreachable. Since earlier in the code `lookup_fut` is defined as

```rust
 let lookup_fut = async move {
                    let res = this.blocks_from_store(hashes).await;
                    BlocksLookupResult::ByHash(Arc::new(res))
                };
```

So it always returns `ByHash`. But i'll just go ahead and return an error since thats defenetly cleaner

Reply:
Addressed in latest commit